src/command_modules/azure-cli-container/azure/cli/command_modules/container/custom.py

```diff
@@ -231,6 +237,63 @@ def container_logs(cmd, resource_group_name, name, container_name=None, follow=F
             stream_args=(logs_client, resource_group_name, name, container_name, container_group.restart_policy))
 
 
+def container_exec(cmd, resource_group_name, name, container_name=None, exec_command=None, terminal_row_size=None, terminal_col_size=None):
+    """Start exec for a container. """
+
+    if platform.system() is WINDOWS_OS_NAME:
+        print("The container exec operation is currently not supported for Windows.")
+        return
+
+    start_container_client = cf_start_container(cmd.cli_ctx)
+    container_group_client = cf_container_groups(cmd.cli_ctx)
+    container_group = container_group_client.get(resource_group_name, name)
+
+    # Set defaults for rows and cols size
+    if terminal_row_size is None:
+        terminal_row_size = 24
+
+    if terminal_col_size is None:
+        terminal_col_size = 80
+
+    # If container name is not present, use the first container.
+    if container_name is None:
+        container_name = container_group.containers[0].name
+
+    terminal_size = ContainerExecRequestTerminalSize(rows=terminal_row_size, cols=terminal_col_size)
+
+    execContainerResponse = start_container_client.launch_exec(resource_group_name, name, container_name, exec_command, terminal_size)
+
+    _start_exec_pipe(execContainerResponse.web_socket_uri, execContainerResponse.password)
+
+
+def _start_exec_pipe(web_socket_uri, password):
+    ws = websocket.create_connection(web_socket_uri)
+    try:
+        ws.send(password)
+
+        while True:
+            try:
+                r, w, e = select.select([ws.sock, sys.stdin], [], [])
+                if ws.sock in r:
+                    data = ws.recv()
+                    if not data:
+                        break
+                    sys.stdout.write(data)
+                    sys.stdout.flush()
+                if sys.stdin in r:
+                    x = sys.stdin.read(1)
+                    if not x:
+                        break
+                    ws.send(x)
+            except (select.error, IOError) as e:
+                if e.args and e.args[0] == errno.EINTR:
+                    pass
+                else:
+                    raise
+    except websocket.WebSocketException:
+        print("Connection Ended.")
```

Comment:
https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md
"Log to logger.error() or logger.warning() for user messages; do not use the print() function"